bin/ocaml_merlin.ml

```diff
@@ -78,14 +78,13 @@ end = struct
   let load_merlin_file file =
     (* We search for an appropriate merlin configuration in the current
        directory and its parents *)
-    let filename = String.lowercase_ascii (Path.Build.basename file) in
```

Comment:
@voodoos why was this needed in the first place?

Reply:
So, the basename is needed (but I see you moved it to the `get` function, because the path queried by Merlin is the path to the source file.

I think it would be clearer to call `lowercase_ascii` only when the comparison with module names is done.
Since module names always start with a capital but source files might not it is necessary to compare them modulo capitalization. But there might always be such a comparison function in Dune ?

Reply:
Right, I get the point of getting the basename, but why are we lowercasing it?

Reply:
The idea was to ease the comparison with the module name but it might be incorrect on case sensitive filesystem.
I guess we should just capitalize (the first letter) of the basename ?

Reply:
I see. Does it not work if we just do case sensitive look ups? Or does merlin try to create filenames from module names?

Reply:
I think we still need to capitalize the first letter since Merlin ask not for the module but from the filename which might or might not be starting with a uppercase letter. Since we compare it to existing module names to find the correct one we need to have the first letter capitalized.

Comparison is done here: https://github.com/ocaml/dune/pull/6594/files#diff-12c5db016a4c965fce3133c10a775ba29bf8ea300342004228ff877721cd5a42R186-R188

But maybe there is a better way to do it ?

Reply:
So we want to go from the filename to the module, right?

What about just building a reverse map from the filename without the extension to the module? That seems like it will work reliably and is something we could use elsewhere as well.

Reply:
You mean keep a table linking module names to filenames ?
Yes, that sounds like a useful thing to have.

Reply:
We already have a table from module names to filenames. We can find the module by name in `Modules.t` and then get the filename from `Module.file`.